src/librustc_typeck/check/wfcheck.rs

```diff
@@ -342,6 +342,78 @@ fn check_trait<'tcx>(tcx: TyCtxt<'tcx, 'tcx>, item: &hir::Item) {
 
     for_item(tcx, item).with_fcx(|fcx, _| {
         check_where_clauses(tcx, fcx, item.span, trait_def_id, None);
+
+        // Type-check associated type defaults (if there are any):
```

Comment:
Would be good to refactor this check into its own function like `check_where_clauses` above so it reads more pseudo-code-y.